Endpoints/player/MapsEndpoint/MapsEndpoint.cs

```diff
@@ -308,6 +307,47 @@ public Model.Maps GetSimple(OLabDBContext context, uint id)
       return dtoList;
     }
 
+    /// <summary>
+    /// Retrieve all sessions for a given map
+    /// </summary>
+    /// <param name="mapId"></param>
+    /// <returns></returns>
+    public async Task<IList<SessionInfo>> GetSessionsAsync(
+      IOLabAuthentication auth,
+      uint mapId)
+    {
+      logger.LogDebug($"{auth.GetUserContext().UserId}: MapsEndpoint.GetSessionsAsync");
+
+      // test if user has access to map.
+      if (!auth.HasAccess("W", Utils.Constants.ScopeLevelMap, mapId))
+        throw new OLabUnauthorizedException(Utils.Constants.ScopeLevelMap, mapId);
+
+      Maps map = GetSimple(dbContext, mapId);
+
+      if (map == null)
+        throw new OLabObjectNotFoundException(Utils.Constants.ScopeLevelMap, mapId);
+
+      IList<SessionInfo> sessions = await dbContext.SessionInfo.FromSqlInterpolated(@$"
+        select s.uuid,
+               cast(from_unixtime(s.start_time) as datetime) as `Timestamp`,
```

Comment:
A hard-coded SQL state should not be necessary if EF relationships between entities are set up correctly.  Please investigate.